src/config/network.rs

```diff
@@ -2,34 +2,43 @@ use std::borrow::Cow;
 
 /// Network configuration.
 #[derive(Debug)]
-pub enum Interface<'i> {
-    /// CNIConfiguration that will be used to generate the VM's network namespace,
-    /// tap device and internal network for this network interface.
-    Cni(Cni<'i>),
+pub struct Interface<'i> {
+    host_if_name: Cow<'i, str>,
+    vm_if_name: Cow<'i, str>,
 }
 
-/// CNI network configuration.
-// TODO: Provide a builder for `Cni`.
-#[derive(Debug, Default)]
-pub struct Cni<'c> {
-    /// Corresponds to the "name" parameter in the CNI spec's
-    /// Network Configuration List structure. It selects the name
-    /// of the network whose configuration will be used when invoking CNI.
-    pub network_name: Cow<'c, str>,
+impl<'i> Interface<'i> {
+    /// Create a new `Interface` instance.
+    pub fn new<H, V>(host_if_name: H, vm_if_name: V) -> Self
```

Comment:
More curiosity why using traits, aren't these likely to be `&str` anyways?

Reply:
The generics here allow use of `&str`, `&'static str` and `String`. If people want to keep the struct around but not care about lifetimes and string allocation, they can just use `String` (or string literals) and `'static` lifetime.

Reply:
doesn't `host_if_name: &str` do the same thing (besides the cow) or is the trait needed to pass in any of those as a cow?

Reply:
> doesn't host_if_name: &str do the same thing (besides the cow)

no, you can't pass `String` to a `&str` arg. You'll need to borrow the string and that would mean the caller has to keep the string alive as long as the config they're adding it to (i-e the config struct can't own the string then).

> is the trait needed to pass in any of those as a cow?

We can have the caller having to `.into()` their strings before passing but this makes the API a lot more convenient since then caller don't have to worry about coversion to `Cow`.